index.js

```diff
@@ -58,12 +43,6 @@ Geocoder.prototype = inherit(mapboxgl.Control, {
   onAdd: function(map) {
     this.request = new XMLHttpRequest();
 
-    this.container = this.options.container ?
-      typeof this.options.container === 'string' ?
-      document.getElementById(this.options.container) :
-      this.options.container :
-      map.getContainer();
```

Comment:
Because the PR is about updating to the 0.27.0 API, and the API defines that you return an element and the map adds that element. Attaching to an arbitrary DOM node is not part of the API.